M/MAGMA/build_tarballs.jl

```diff
@@ -1,56 +1,38 @@
-using BinaryBuilder
+using BinaryBuilder, Pkg
 
-# Collection of sources required to build Nettle
 name = "MAGMA"
-version = v"2.5.1"
+version = v"2.6.2"
 sources = [
-    "http://icl.utk.edu/projectsfiles/magma/downloads/magma-2.5.1.tar.gz" =>
-    "ce32c199131515336b30c92a907effe0c441ebc5c5bdb255e4b06b2508de109f",
-#    "./bundled",
+    GitSource("https://bitbucket.org/icl/magma", "864fa8b28e8c2905a41fbef32f156695378dc104"),
+    DirectorySource("./bundled")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/magma-*/
+cd $WORKSPACE/srcdir/magma/
 
-# We need a newer version of FindCUDA.cmake
-curl -L 'https://raw.githubusercontent.com/Kitware/CMake/v3.13.0/Modules/FindCUDA.cmake' -o /usr/share/cmake/Modules/FindCUDA.cmake
+cp ../make.inc .
 
-# Apply patches to force name-mangling
-atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fortran_mangling.patch
-
-mkdir build && cd build
-export CFLAGS="${CFLAGS} -DMAGMA_ILP64"
-cmake .. -DCMAKE_TOOLCHAIN_FILE="${CMAKE_TARGET_TOOLCHAIN}" \
-    -DLAPACK_LIBRARIES="-lopenblas64_ -lgfortran -lquadmath" \
-    -DCUDA_TOOLKIT_ROOT_DIR="${prefix}" \
-    -DCUDA_TOOLKIT_TARGET_DIR="${prefix}" \
-    -DGPU_TARGET="Maxwell Pascal Volta" \
-    -DCUDA_TOOLKIT_ROOT_DIR_INTERNAL="${prefix}"
-
-make -j${nproc}
-make install
+make -j${nproc} sparse-lib
+make install prefix=${prefix}
+install_license COPYRIGHT
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = [
-    Platform("x86_64", "linux"),
-    Platform("x86_64", "windows"),
-    Platform("x86_64", "macos"),
-]
-platforms = expand_gcc_versions(platforms)
+platforms = [Platform("x86_64", "linux"; libc="glibc")]
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libMAGMA", :libMAGMA),
+    LibraryProduct("libmagma", :libmagma),
+    LibraryProduct("libmagma_sparse", :libmagma_sparse)
```

Comment:
Libraries needing CUDA generally shouldn't be dlopened automatically (otherwise that'd fail on systems without cuda): https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.LibraryProduct
```suggestion
    LibraryProduct("libmagma", :libmagma; dont_dlopen=true),
    LibraryProduct("libmagma_sparse", :libmagma_sparse; dont_dlopen=true)
```